internal/cloudrunci/cloudrunci.go

```diff
@@ -154,16 +154,10 @@ func (s *Service) Request(method string, path string, opts ...func(*RetryOptions
 	var lastSeen error
 	resp := &http.Response{}
 	for i := 0; i < options.MaxAttempts; i++ {
-		req, err := s.NewRequest(method, path)
-		if err != nil {
-			lastSeen = err
-			continue
-		}
 		defaultClient := &http.Client{}
 
-		resp, err = defaultClient.Do(req)
-		if err != nil {
-			lastSeen = err
+		resp, lastSeen = defaultClient.Do(req)
+		if lastSeen != nil {
 			continue
 		}
 		if options.ShouldAccept(resp) {
```

Comment:
suggestion: GitHub UI is fighting with me, but here's another bit of review on previously existing code: a simplification opportunity.

```go
if options.ShouldAccept(resp) {
    return resp, nil
}
time.Sleep(options.Delay)
```

1. No need for the else clause because all code will return before reaching it or execute the clause.
2. No need for continue since all code will continue if it reaches the sleep.

Reply:
Good call, thanks!